basics/units.ipynb

```diff
@@ -0,0 +1,403 @@
+    "| ------------------ :| ----------------- :| -- :|\n",
+    "|  Length             |  millimeter        | mm  |\n",
+    "|  Time               |  nanosecond        | ns  |\n",
+    "|  Energy             |  Mega electron Volt| MeV |\n",
+    "|  Positron charge    |  eplus             |     |\n",
+    "|  Temperature        |  kelvin            | K   |\n",
+    "|  Amount of substance|  mole              | mol |\n",
+    "|  Luminous intensity |  candela           | cd  |\n",
+    "| Plane angle         |  radian            | rad |\n",
+    "|  Solid angle        |  steradian         | sr  |\n",
+    "\n",
+    "Note: no need to make use of sophisticated packages (e.g. as in AstroPy) since we basically never need to change systems of units (we never use ergs as energy, for example ;-))."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### **Basic usage**\n",
+    "\n",
+    "**Basic usage is straightforward**, though it may be confusing at first. Remember, all variables are written wrt to the units:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "True"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from hepunits import mm, ns, MeV, eplus, GeV, kelvin, mol, cd, rad, sr\n",
+    "\n",
+    "mm == ns == MeV == eplus == kelvin == mol == cd == rad == sr == 1"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "True"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "GeV == 1000*MeV"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Add two quantities with different length units:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "1050.0"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from hepunits import units as u\n",
+    "\n",
+    "1*u.meter + 5*u.cm"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The result is in HEP units, so mm. Indeed, for example `u.meter == 1000 == 1000 * u.mm`. \n",
+    "\n",
+    "Rather obtain the result in meters:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "1.05"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "(1*u.meter + 5*u.cm) / u.meter"
```

Comment:
I think it is worth mentioning explicitly here already, that the system does not protect against unit errors. For example, the following does not raise an error
```
1 * u.second + 1 * u.meter
```

Reply:
Maybe also give a short explanation why this is a deliberate choice. It is comparably easy to write a system that finds these errors for normal numbers, but then you also need numpy arrays with units and all numpy ufuncs must work with those and so forth. Such extended arrays do not work with numba or wrapped C code, etc.

Reply:
I think the main benefit of using units is that the LHCb software and Scikit-HEP uses the same units, therefore you can do
```
from particle import literals as lp
# gimme mass in GeV
lp.pi_plus.mass / u.GeV
```

Reply:
Fully agree. Readability and consistency with the HEP ecosystem should be advertised. Tt may be worth pointing out in a FAQ section and linking to it or just somewhere in the README, to answer mainly the question: "Why aren't units checked/error propagated/nice title/printouts"

Reply:
> I think the main benefit of using units is that the LHCb software and Scikit-HEP uses the same units …

In fact it is far more than Scikit-HEP, and that's central: it's Geant4, CLHEP, Gaudi and many others. That is why I always say "the HEP system of units".

Reply:
> Fully agree. Readability and consistency with the HEP ecosystem should be advertised. It may be worth pointing out in a FAQ section and linking to it or just somewhere in the README, to answer mainly the question: "Why aren't units checked/error propagated/nice title/printouts"

Have you got a suggestion? I take the very good points from you both and realise that the `hepunits` README needs this kind of information.
In the meantime I made some updates. Will try and do a bit more. Thank you.